fxa-oauth-server/lib/routes/key_data.js

```diff
@@ -48,40 +50,47 @@ module.exports = {
     const requestedClientId = req.payload.client_id;
 
     return P.all([
-      verify(req.payload.assertion),
+      verifyAssertion(req.payload.assertion),
       db.getClient(Buffer.from(requestedClientId, 'hex')).then((client) => {
         if (client) {
           // find all requested scopes that are allowed for this client.
           const allowedScopes = ScopeSet.fromString(client.allowedScopes || '');
           const scopeLookups = requestedScopes.filtered(allowedScopes).getScopeValues().map(s => db.getScope(s));
           return P.all(scopeLookups).then((result) => {
-            return result.filter((s) => !! (s && s.hasScopedKeys));
+            return result.filter((s) => !! (s && s.hasScopedKeys))
+              .map(s => {
+                // When we implement key rotation these values will come from the db.
+                // For now all scoped keys have the default values.
+                s.keyRotationSecret = DEFAULT_KEY_ROTATION_SECRET;
+                s.keyRotationTimestamp = DEFAULT_KEY_ROTATION_TIMESTAMP;
+                return s;
+              });
           });
         } else {
           logger.debug('keyDataRoute.clientNotFound', { id: req.payload.client_id });
           throw AppError.unknownClient(requestedClientId);
         }
       })
-    ]).then((results => {
-      logger.debug('keyDataRoute.results', JSON.stringify(results));
-      const assertionData = results[0];
-      const scopeResults = results[1];
+    ]).spread((claims, scopes) => {
+      const iat = claims.iat || claims['fxa-lastAuthAt'];
       const response = {};
-
-      if (assertionData['fxa-lastAuthAt'] < (Date.now() - AUTH_EXPIRES_AFTER_MS) / 1000) {
-        throw AppError.staleAuthAt(assertionData['fxa-lastAuthAt']);
-      }
```

Comment:
We don't need this check, it was added as an extra sanity-check when we assumed that every scoped-key login flow would have to be accompanied by the user entering their password.  That's no longer true with the desktop-based pairing flow, because it will already have the necessary key material.  So I think we can safely remove it.

Reply:
Yes, `exp` is still enforced with its normal semantics.  "Stale" in this context means something like "the assertion was valid according to normal BrowserID/JWT validation rules, but we did some additional checks on the timestamps that caused us to reject it".

An assertion with `exp` in the past would not be "stale", it would be just straight-up invalid.

Perhaps we should introduce a new error code for this rather than re-using `staleAuthAt`, since this is not actually checking the last-auth-at timestamp?  We could use a dedicated error for "this assertion was generated prior to a key-rotation event, you need to generate a fresh one".